spyder/api/widgets/menus.py

```diff
@@ -272,6 +308,41 @@ def _add_section(self, section, before_section=None):
                 idx = idx if (idx == 0) else (idx - 1)
                 self._sections.insert(idx, after_section)
 
+    def _set_icons(self):
+        """
+        Unset menu icons for app menus and set them for regular menus.
+
+        This is necessary only for Mac to follow its Human Interface
+        Guidelines (HIG), which don't recommend icons in app menus.
+        """
+        if sys.platform == "darwin":
```

Comment:
This method can also be called directly (i.e. not as a result of `aboutToShow`), so that's why this `if` is also necessary.

And example of that is the call that `console_environment_menu` needs to do to show icons for its items.